File: auth0/src/test/java/com/auth0/android/request/internal/ResponseUtilsTest.java
package com.auth0.android.request.internal;


import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;

import okhttp3.ResponseBody;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class ResponseUtilsTest {

    @Rule
    public ExpectedException exception = ExpectedException.none();

    @Test
    public void shouldCloseBody() throws Exception {
        ResponseBody body = mock(ResponseBody.class);
        ResponseUtils.closeStream(body);

        verify(body).close();
    }
}

File: auth0/src/test/java/com/auth0/android/request/internal/ResponseUtilsTest.kt
package com.auth0.android.request.internal

import junit.framework.TestCase.assertTrue
import org.hamcrest.CoreMatchers
import org.hamcrest.MatcherAssert
import org.junit.Test
import java.net.SocketException
import java.net.SocketTimeoutException
import java.net.UnknownHostException

public class ResponseUtilsTest {

    @Test
    public fun testIsNetworkErrorWhenSocketExceptionOccurs() {
        MatcherAssert.assertThat(ResponseUtils.isNetworkError(SocketException()), CoreMatchers.`is`(true))
    }

    @Test
    public fun testIsNetworkErrorWhenSocketTimeoutExceptionOccurs() {
        MatcherAssert.assertThat(ResponseUtils.isNetworkError(SocketTimeoutException()), CoreMatchers.`is`(true))
    }

    @Test
    public fun testIsNetworkErrorWhenUnknownHostExceptionOccurs() {
        MatcherAssert.assertThat(ResponseUtils.isNetworkError(UnknownHostException()), CoreMatchers.`is`(true))
    }
}